### PR TITLE
Support output node documentation

### PR DESCRIPTION
**Pull request by ke-yu**
### Purpose

This PR is to support output node documentation in custom workspace.

Just like input node [enhancement](https://github.com/DynamoDS/Dynamo/pull/5676), output node is updated to support comments, and comments will be displayed as tool-tip in custom node's corresponding output port.

For example:

And tool-tip:

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.

### Reviewer

@aparajit-pratap 

### FYIs

@mostaphaRoudsari, the input node issue you mentioned in #5676 also fixed here.

### CODE REVIEW

**Comment by mostaphaRoudsari** [comment]
Thanks @ke-yu! This looks great. I have been using descriptions for all the inputs and it's very helpful. There is one minor issue that you may want to fix to make it perfect. Currently if I use `:` in the description it will also break the parsing. Here is an example:

image

image

File: src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs [hunk 1]
```diff
@@ -393,11 +396,26 @@ protected override void DeserializeCore(XmlElement nodeElement, SaveContext cont
...
+                    // "x:int" will be compiled to "x:int = tTypedIdent0";
+                    else if (rightIdent != null && rightIdent.Value.StartsWith(Constants.kTempVarForTypedIdentifier))
+                    {
+                        identifier = leftIdent;
+                    }
+                    else
+                    {
+                        identifier = leftIdent;
+                    }
+
```
Comment (aparajit-pratap) [inline src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs]: All the above changes look related to `Input` node. What defect is this fixing?

File: src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs [hunk 2]
```diff
@@ -514,5 +572,69 @@ protected override bool UpdateValueCore(UpdateValueParams updateValueParams)
...
+                        this.Error(Properties.Resources.WarningInvalidOutput);
+                    }
+                }
+                else
+                {
+                    var leftIdent = node.LeftNode as IdentifierNode;
+                    var rightIdent = node.RightNode as IdentifierNode;
+
+                    // "x" will be compiled to "temp_guid = x;"
+                    if (leftIdent != null && leftIdent.Value.StartsWith(Constants.kTempVarForNonAssignment))
```
Comment (aparajit-pratap) [inline src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs]: Thanks for the comments!

File: src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs [hunk 3]
```diff
@@ -514,5 +572,69 @@ protected override bool UpdateValueCore(UpdateValueParams updateValueParams)
...
+                    {
+                        if (parseParam.Errors.Any())
+                        {
+                            this.Error(parseParam.Errors.First().Message);
+                        }
+                        else
+                        {
+                            this.Warning(Properties.Resources.WarningInvalidOutput);
+                        }
+                        outputIdentifier = leftIdent;
```
Comment (aparajit-pratap) [inline src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs]: Can't seem to imagine a case where there is an invalid output and you still want to assign the left identifier to the output. Can you give an example?

**Comment by aparajit-pratap** [comment]
@ke-yu this looks good aside from a few questions I have above. Do you have any test cases?

File: src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs (hunk 1 above)
Comment (ke-yu) [inline src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs]: Issue described in #5676.

**Comment by ke-yu** [comment]
@aparajit-pratap test case added.

**Comment by ke-yu** [comment]
@mostaphaRoudsari thanks for reporting this issue. Fixed in this PR.

File: src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs (hunk 3 above)
Comment (ke-yu) [inline src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs]: if output is invalid, we still try to display whatever string in the output port... so technically the variable name 'outputIdentifier' is not quite correct...

**Comment by aparajit-pratap** [comment]
Thanks, LGTM!

**Pull request merged**